Source/JavaScriptCore/bytecode/CodeBlock.cpp

```diff
@@ -852,12 +852,11 @@ CodeBlock::~CodeBlock()
         vm.m_perBytecodeProfiler->notifyDestruction(this);
 
     if (LIKELY(!vm.heap.isShuttingDown())) {
-        if (m_metadata) {
+        // FIXME: This check should really not be necessary, see https://webkit.org/b/272787
+        ASSERT(!m_metadata || m_metadata->unlinkedMetadata());
+        if (m_metadata && !m_metadata->isDestroyed()) {
```

Comment:
Same here in order to mitigate the unexpected case that the `MetadataTable` was valid but the `unlinkedMetadata` was gone.

Reply:
We saw the case where `m_metadata` was not null, but the `unlinkedMetadata` was null.  It is unexplained as to how this happens.  Hence, this is a speculative fix.